frontend/appflowy_flutter/lib/user/application/reminder/reminder_bloc.dart

```diff
@@ -452,55 +454,84 @@ class ReminderBloc extends Bloc<ReminderEvent, ReminderState> {
+      return true;
+    } else {
+      /// blockId is null means no node
+      final blockId = reminder.meta[ReminderMetaKeys.blockId];
+      if (blockId == null) {
+        removeIds.add(reminder.id);
+        return false;
       }
-      if (current.children.isNotEmpty) {
-        for (final child in current.children) {
-          final node = searchById(child, id);
 
-          if (node != null) {
-            return node;
+      /// check if document is not null
+      final document = await DocumentService()
+          .openDocument(documentId: viewId)
+          .fold((s) => s.toDocument(), (_) => null);
+      if (document == null) {
+        removeIds.add(reminder.id);
+        return false;
+      }
+      Node? searchById(Node current, String id) {
+        if (current.id == id) {
+          return current;
+        }
+        if (current.children.isNotEmpty) {
+          for (final child in current.children) {
+            final node = searchById(child, id);
+
+            if (node != null) {
+              return node;
+            }
           }
         }
+        return null;
       }
-      return null;
-    }
 
-    /// check if node is not null
-    final node = searchById(document.root, blockId);
-    if (node == null) return false;
-    final textInserts = node.delta?.whereType<TextInsert>();
-    if (textInserts == null) return false;
-    for (final text in textInserts) {
-      final mention =
-          text.attributes?[MentionBlockKeys.mention] as Map<String, dynamic>?;
-      final reminderId = mention?[MentionBlockKeys.reminderId] as String?;
-      if (reminderIds.contains(reminderId)) {
-        return true;
+      /// check if node is not null
+      final node = searchById(document.root, blockId);
+      if (node == null) {
+        removeIds.add(reminder.id);
+        return false;
       }
+      final textInserts = node.delta?.whereType<TextInsert>();
+      if (textInserts == null) return false;
+      for (final text in textInserts) {
+        final mention =
+            text.attributes?[MentionBlockKeys.mention] as Map<String, dynamic>?;
+        final reminderId = mention?[MentionBlockKeys.reminderId] as String?;
+        if (reminderIds.contains(reminderId)) {
+          return true;
+        }
+      }
+      removeIds.add(reminder.id);
+      return false;
     }
-    return false;
   }
 
   Future<List<ReminderPB>> filterAvailableReminders(
-    List<ReminderPB> reminders,
-  ) async {
+    List<ReminderPB> reminders, {
+    bool removeUnavailableReminder = false,
+  }) async {
     final List<ReminderPB> availableReminders = [];
     final reminderIds = reminders.map((e) => e.id).toSet();
+    final removeIds = <String>{};
     for (final r in reminders) {
-      if (await checkReminderAvailable(r, reminderIds)) {
+      if (await checkReminderAvailable(r, reminderIds, removeIds: removeIds)) {
         availableReminders.add(r);
       }
     }
+    if (removeUnavailableReminder) {
```

Comment:
Removing a reminder will trigger a refresh. Since you've added auto refresh logic, you can remove the refresh logic from the removeReminder event.